pkg/manager/impl/util/shared_test.go

```diff
@@ -474,3 +476,64 @@ func TestListActiveLaunchPlanVersionsFilters(t *testing.T) {
 	assert.Equal(t, activeExpr.Args, int32(admin.LaunchPlanState_ACTIVE))
 	assert.Equal(t, activeExpr.Query, testutils.StateQueryPattern)
 }
+
+func TestGetMatchableResource(t *testing.T) {
+	resourceType := admin.MatchableResource_WORKFLOW_EXECUTION_CONFIG
+	project := "dummyProject"
+	domain := "dummyDomain"
+	t.Run("successful fetch", func(t *testing.T) {
+		resourceManager := &managerMocks.MockResourceManager{}
+		resourceManager.GetResourceFunc = func(ctx context.Context,
+			request managerInterfaces.ResourceRequest) (*managerInterfaces.ResourceResponse, error) {
+			assert.EqualValues(t, request, managerInterfaces.ResourceRequest{
+				Project:      project,
+				Domain:       domain,
+				ResourceType: resourceType,
+			})
+			return &managerInterfaces.ResourceResponse{
+				Attributes: &admin.MatchingAttributes{
+					Target: &admin.MatchingAttributes_WorkflowExecutionConfig{
+						WorkflowExecutionConfig: &admin.WorkflowExecutionConfig{
+							MaxParallelism: 12,
+						},
+					},
+				},
+			}, nil
+		}
+
+		_, err := GetMatchableResource(context.Background(), resourceManager, resourceType, project, domain)
```

Comment:
nit: also check that the expected resource is returned?

Reply:
Done .